typing/parmatch.ml

```diff
@@ -1656,7 +1656,7 @@ let rec every_satisfiables pss qs = match qs.active with
         let pss = simplify_first_col pss in
         (* The handling of incoherent matrices is kept in line with
            [satisfiable] *)
-        if not (first_column_is_coherent pss) then
+        if not (first_column_is_coherent ((uq, make_row rem) :: pss)) then
```

Comment:
maybe we could write `(all_coherent (first_column pss))` and `(all_coherent (uq :: first_column pss))` to avoid having to come up with the rest of the row that we don't actually care about?

Reply:
Ah, that's what you meant by "the API might need some adjusting"!

Yes, I too find my current implementation a bit distasteful and yours seems prettier.  But I'm also worried that yours means iterating over the rows once more, and allocating a fresh list with as many elements as there are rows, whereas with the current implementation we don't need to iterate and don't allocate much.

I'm tempted to keep my implementation because of these (probably unwarranted) performance concerns. But I'll welcome your input on that issue and am ready to change my mind if you feel differently.

Reply:
I do think that the performance concern is unwarranted (the compiler already allocates, this case of the function already allocates, and we have no latency constraints), but from your answers regarding `matrix_stable_vars` it looks like I should send a cleanup PR after this one is merged anyway, so I don't mind proposing some API changes at the same time.

Reply:
Alright, sounds good to me :)
I will leave things as is for now then.